tensorflow/compiler/xla/service/algebraic_simplifier.cc

```diff
@@ -166,6 +166,61 @@ bool IsUnstridedSlice(const HloInstruction* hlo) {
                         [](int64 stride) { return stride == 1; });
 }
 
+// Returns bool to determine whether a pair of converts can be eliminated
+bool IsConvertPairNoOp(const HloInstruction* convert) {
+  //                       [operand_convert]          [convert]
+  // (operand_convert_input)->convert-(convert_input)->convert-(dest)
+
+  const HloInstruction* operand_convert = convert->operand(0);
```

Comment:
I'd be a bit paranoid here and add `CHECK` that the inner instruction is also a `kConvert`.